gandi/resource_glue_record.go

```diff
@@ -0,0 +1,150 @@
+package gandi
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/go-gandi/go-gandi/domain"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceGlueRecord() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGlueRecordCreate,
+		Read:   resourceGlueRecordRead,
+		Update: resourceGlueRecordUpdate,
+		Delete: resourceGlueRecordDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"zone": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Domain name",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Host name of the glue record.",
+			},
+			"values": {
+				Type:        schema.TypeList,
+				Required:    true,
+				ForceNew:    true,
+				Description: "List of IP addresses.",
+			},
+		},
+		Timeouts: &schema.ResourceTimeout{Default: schema.DefaultTimeout(1 * time.Minute)},
+	}
+}
+
+func resourceGlueRecordCreate(d *schema.ResourceData, meta interface{}) (err error) {
+	client := meta.(*clients).Domain
+	resDomain := d.Get("zone").(string)
+	name := d.Get("name").(string)
+
+	var values []string
+	for _, i := range d.Get("values").([]interface{}) {
+		values = append(values, i.(string))
+	}
+	sort.Strings(values)
+
+	request := domain.GlueRecordCreateRequest{
+		Name: name,
+		IPs:  values,
+	}
+
+	err = client.CreateGlueRecord(resDomain, request)
+	if err != nil {
+		return
+	}
+
+	// Sent, got 202 response. Delay read.
+	time.Sleep(2 * time.Second)
```

Comment:
You have to use a ["retry"](https://www.terraform.io/docs/extend/resources/retries-and-customizable-timeouts.html#retry) instead.

Reply:
Ok, will update.  Followed the example of [this resource](https://github.com/go-gandi/terraform-provider-gandi/blob/master/gandi/resource_dnsseckey.go#L66-L67) - I suppose that should be updated?

With how this retry will work, I will also defer until I have the Get request available within the go client.

Reply:
Updated.  Question around the lack of retry for dns key resource remains.